content-repo/gen_top_contrib.py

```diff
@@ -0,0 +1,214 @@
+    Args:
+        prs: list of internal prs.
+
+    Returns:
+        inner_prs: list of external prs.
+        pr_bodies: list of external pr bodies.
+    """
+    inner_prs = []
+    pr_bodies = []
+    for pr in prs:
+        pr_body = pr.get('body', '')
+        inner_pr = PR_NUMBER_REGEX.search(pr_body)
+        if inner_pr is not None:
+            pr_bodies.append(pr_body)
+            inner_prs.append(inner_pr[0])
+
+    return pr_bodies, inner_prs
+
+
+def github_pagination_prs(url: str, params: dict, res):
+    """
+    Paginate throw all the pages in Github according to search query
+    Args:
+        url (str): the request url
+        params (dict): params for the request
+        res: the response from the http_request
+
+    Returns: prs (dict)
+    """
+    prs = []
+    link = res.headers.get('Link')
+    if link:
+        links = link.split(',')
+        for link in links:
+            last_page = link[link.find("<")+1:link.find(">")][-1]
+        while params['page'] <= int(last_page):
+            params['page'] = params['page'] + 1
+            response = requests.request('GET', url, params=params, headers=HEADERS, verify=False)
+            next_page_prs = response.json().get('items', [])
+            prs.extend(next_page_prs)
+
+    return prs
+
+
+def get_contractors_prs():
+    url = URL + '/search/issues'
+    query = 'type:pr state:closed org:demisto repo:content is:merged base:master ' \
+            'head:contrib/crest head:contrib/qmasters head:contrib/mchasepan'
+    params = {
+        'q': query,
+        'per_page': 100,
+        'page': 1
+    }
+    res = requests.request('GET', url, headers=HEADERS, params=params, verify=False)
+    prs = res.json().get('items', [])
+
+    next_pages_prs = github_pagination_prs(url, params, res)
+    prs.extend(next_pages_prs)
+
+    _, inner_prs = get_external_prs(prs)
+
+    return inner_prs
+
+
+def get_contrib_prs():
+
+    url = URL + '/search/issues'
+    query = 'type:pr state:closed org:demisto repo:content is:merged base:master head:contrib/'
+    params = {
+        'q': query,
+        'per_page': 100,
+        'page': 1
+    }
+    res = requests.request('GET', url, headers=HEADERS, params=params, verify=False)
+    prs = res.json().get('items', [])
+
+    # Get all the PRs from all pages
+    next_pages_prs = github_pagination_prs(url, params, res)
+    prs.extend(next_pages_prs)
+
+    pr_bodies, inner_prs = get_external_prs(prs)
+    contractors_prs = get_contractors_prs()
+    for pr in inner_prs:
+        if pr in contractors_prs:
+            inner_prs.remove(pr)
+
+    return pr_bodies, inner_prs
+
+
+def get_github_user(user_name):
+    url = f'{URL}/users/{user_name}'
+    res = requests.request('GET', url, headers=HEADERS, verify=False)
+    response = res.json()
+
+    github_avatar = response.get('avatar_url')
+    github_profile = response.get('html_url')
+
+    return github_avatar, github_profile
+
+
+def get_pr_user():
```

Comment:
lets split this function into 2: one to retrieve the users and one to do the manipulation on the result of the http response